instrumentation/spring/spring-cloud-gateway/spring-cloud-gateway-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/v2_0/ServerWebExchangeHelper.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.v2_0;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.web.server.ServerWebExchange;
+
+public final class ServerWebExchangeHelper {
+
+  /** Route info key. */
+  public static final String ROUTE_INFO_ATTRIBUTES = "ROUTE_INFO";
+
+  private ServerWebExchangeHelper() {}
+
+  public static void extractAttributes(ServerWebExchange exchange, Context context) {
+    // Record route info
+    Route route = exchange.getAttribute(GATEWAY_ROUTE_ATTR);
+    if (route != null) {
+      Span currentSpan = Span.fromContext(context);
+      assert currentSpan != null;
+      currentSpan.setAttribute(ROUTE_INFO_ATTRIBUTES, route.toString());
```

Comment:
Typically by default we only add attributes that are defined in the spec, other attributes are added only when enabled with a flag. Search for `experimental-span-attributes` under instrumentation for samples.
Also the content of this attribute is `ROUTE_INFO=Route{id='path_route', uri=h1c://mock.response, order=0, predicate=org.springframework.cloud.gateway.support.ServerWebExchangeUtils$$Lambda$1000/0x000000080175d8c8@8d25799, gatewayFilters=[OrderedGatewayFilter{delegate=io.opentelemetry.javaagent.instrumentation.spring.gateway.v2_0.GatewayTestApplication$$Lambda$1002/0x000000080175dfe0@1f5505d9, order=0}]}}`. The lambdas here don't look useful.